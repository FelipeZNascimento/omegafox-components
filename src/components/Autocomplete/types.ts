export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
