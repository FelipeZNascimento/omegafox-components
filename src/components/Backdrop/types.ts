export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
