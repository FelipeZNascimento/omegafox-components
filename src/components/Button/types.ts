export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
