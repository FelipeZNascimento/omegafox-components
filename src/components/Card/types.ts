export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
