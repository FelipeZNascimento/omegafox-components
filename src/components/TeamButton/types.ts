export type TBorderPosition =
  | 'bottomLeft'
  | 'bottomRight'
  | 'topLeft'
  | 'topRight';

export interface ITeamButtonProps {
  borderPosition?: TBorderPosition;
  colors: string[];
  fontSize?: 'small' | 'big';
  isBig?: boolean;
  isForceMobile?: boolean;
  isHoverable?: boolean;
  isSelected?: boolean;
  logo: string;
  name: string;
  nameShort: string;
  onClick?: null | (() => void);
}
