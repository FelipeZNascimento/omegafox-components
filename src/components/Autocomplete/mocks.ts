import { TDropdownItem } from './types';

export const dropdownList: TDropdownItem[] = [
  {
    id: 0,
    name: 'Lionel Messi',
    abbreviationEn: 'ARG'
  },
  {
    id: 1,
    name: 'Cristiano Ronaldo',
    abbreviationEn: 'POR'
  },
  {
    id: 2,
    name: 'Mbappé',
    abbreviationEn: 'FRA'
  },
  {
    id: 3,
    name: 'Caio Jr.',
    abbreviationEn: 'BRA'
  },
  {
    id: 4,
    name: 'Darwin Nuñez',
    abbreviationEn: 'URU'
  }
];
