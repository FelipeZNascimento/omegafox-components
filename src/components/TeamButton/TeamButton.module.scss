@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/global.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  @include font-sized(l);
  @include marged(horizontal, s3);
  @include flex-aligned(flex-start, center);

  border-radius: $border-radius;
  flex: 1;
  overflow: hidden;
  position: relative;
  transition: filter $transition-time, box-shadow $transition-time;

  &Selected {
    @include shadowed($color-black, 1);
  }

  &Hoverable {
    cursor: pointer;

    &:hover {
      filter: brightness(110%);
    }
  }

  &Big {
    height: 60px;
  }

  &Small {
    height: 45px;
  }

  &BottomLeft {
    border-bottom-left-radius: 0;
  }
  &BottomRight {
    border-bottom-right-radius: 0;
  }
  &TopRight {
    border-top-right-radius: 0;
  }
  &TopLeft {
    border-top-left-radius: 0;
  }
}

.logoContainer {
  flex: 1;

  .logo {
    filter: drop-shadow(0px 0px 10px #222);
    left: 15%;
    max-height: 50px;
    max-width: 50px;
    position: absolute;
    top: 50%;
    transform: translateX(-50%) translateY(-50%);
  }
}

.nameContainer {
  @include padded(horizontal, m);
  z-index: $absolute-visible;
}

.name {
  &Big {
    @include font-sized(l2);
  }
  &Small {
    @include font-sized(s);
    @include padded(left, l4);

    font-size: 14px;
    text-align: right;
  }
}
