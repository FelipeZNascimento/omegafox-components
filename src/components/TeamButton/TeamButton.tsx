import React from 'react';
import { isMobile } from 'react-device-detect';

import classNames from 'classnames';
import { ITeamButtonProps } from './types';
import styles from './TeamButton.module.scss';
import matchLayer from '../../img/match_layer.png';
import { Tooltip } from 'index';

export const TeamButton = ({
  colors,
  isBig = true,
  isForceMobile,
  isHoverable = true,
  isSelected = false,
  logo,
  name,
  nameShort,
  onClick = null
}: ITeamButtonProps) => {
  const renderLogo = () => {
    return (
      <div className={styles.logoContainer}>
        <img className={styles.logo} alt="logo" src={logo} />
      </div>
    );
  };

  const handleOnClick = () => {
    if (!onClick) {
      return;
    }

    onClick();
  };

  const containerClass = classNames(styles.container, {
    [styles.containerHoverable]: isHoverable,
    [styles.containerSelected]: isSelected,
    [styles.containerBig]: isBig,
    [styles.containerSmall]: !isBig
  });

  return (
    <div
      className={containerClass}
      style={{
        color: colors[0],
        background: `url(${matchLayer}) ${colors[1]}`
      }}
      onClick={handleOnClick}
    >
      {renderLogo()}
      <div
        className={styles.nameContainer}
        style={{
          textShadow: `-1px 0 ${colors[1]}, 0 1px ${colors[1]}, 1px 0 ${colors[1]}, 0 -1px ${colors[1]}`
        }}
      >
        {isMobile || isForceMobile ? (
          <Tooltip text={name}>
            <span>{nameShort}</span>
          </Tooltip>
        ) : (
          name
        )}
      </div>
    </div>
  );
};
