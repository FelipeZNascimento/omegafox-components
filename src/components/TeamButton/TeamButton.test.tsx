import React from 'react';
import { render } from '@testing-library/react';
import { TeamButton } from './TeamButton';
import { ITeamButtonProps } from './types';
import { teamLeft } from '../mocks';

describe('<TeamButton />', () => {
  const renderComponent = ({
    colors,
    isBig,
    isForceMobile,
    logo,
    name,
    nameShort,
    onClick = null
  }: ITeamButtonProps) =>
    render(
      <TeamButton
        colors={colors}
        isBig={isBig}
        isForceMobile={isForceMobile}
        logo={logo}
        name={name}
        nameShort={nameShort}
        onClick={onClick}
      />
    );

  it('should render', () => {
    const { container } = renderComponent({
      colors: teamLeft.colors,
      logo: teamLeft.logo,
      name: teamLeft.name,
      nameShort: teamLeft.nameShort,
      onClick: () => alert('Click!')
    });

    expect(container).toMatchSnapshot();
  });
});
