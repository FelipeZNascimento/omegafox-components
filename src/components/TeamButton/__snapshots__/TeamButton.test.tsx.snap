// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamButton /> should render 1`] = `
<div>
  <div
    class="container containerBottomLeft containerHoverable containerBig"
    style="color: rgb(25, 51, 117); background: rgb(255, 220, 2) url();"
  >
    <div
      class="logoContainer"
    >
      <img
        alt="logo"
        class="logo"
        src="https://assets.omegafox.me/img/countries_crests/bra.png"
      />
    </div>
    <div
      class="nameContainer"
      style="text-shadow: -1px 0 #FFDC02, 0 1px #FFDC02, 1px 0 #FFDC02, 0 -1px #FFDC02;"
    >
      <div
        class="nameBig"
      >
        Brasil
      </div>
    </div>
  </div>
</div>
`;
